From f4b60905abd986c6026e9a75c9b746cc66925436 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Wed, 28 Aug 2013 11:04:25 +0800
Subject: [PATCH 12/25] ENGR00276810 Wifi:refactor wifi ditect function

Refactor wifi detect function to support max 4 slots which should
satisfy all slots on our boards(sabresd, sabreauto,evk, etc)

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 wifi/wifi_unite.c | 30 ++++++++++++++++++++----------
 1 file changed, 20 insertions(+), 10 deletions(-)

diff --git a/wifi/wifi_unite.c b/wifi/wifi_unite.c
index 38b33c5..e2f1041 100644
--- a/wifi/wifi_unite.c
+++ b/wifi/wifi_unite.c
@@ -171,21 +171,31 @@ static char supplicant_prop_name[PROPERTY_KEY_MAX];
 int get_wifi_vendor_info()
 {
 #define SYS_SDIO_BUS_BASE_PATH    "/sys/bus/sdio/devices/"
-#define SYS_SDIO_BUS_MMC1_PATH    SYS_SDIO_BUS_BASE_PATH"mmc1:0001:1/vendor"
-#define SYS_SDIO_BUS_MMC2_PATH    SYS_SDIO_BUS_BASE_PATH"mmc2:0001:1/vendor"
-
+#define SYS_SDIO_BUS_MMC_PATH(num)    SYS_SDIO_BUS_BASE_PATH"mmc"#num":0001:1/vendor"
+#define MAX_SDIO_NUM 4
+    const char* sdio_path[] = {
+        SYS_SDIO_BUS_MMC_PATH(0),
+        SYS_SDIO_BUS_MMC_PATH(1),
+        SYS_SDIO_BUS_MMC_PATH(2),
+        SYS_SDIO_BUS_MMC_PATH(3),
+    };
     char linebuf[1024];
     long value;
-    FILE *f = fopen(SYS_SDIO_BUS_MMC1_PATH, "r");
-    if (!f){
-        ALOGE("Unable to read %s: %s\n", SYS_SDIO_BUS_MMC1_PATH, strerror(errno));
-        f = fopen(SYS_SDIO_BUS_MMC2_PATH, "r");
-        if (!f)
+    int i;
+    FILE *f = NULL;
+    for (i = 0; i < MAX_SDIO_NUM; i++)
+    {
+        f = fopen(sdio_path[i], "r");
+        if (f)
         {
-            ALOGE("Unable to read %s: %s\n", SYS_SDIO_BUS_MMC2_PATH, strerror(errno));
-            return -1;
+            ALOGE("read a device on sdio bus: %s\n", sdio_path[i]);
+            break;
+        } else{
+            ALOGE("unable to read device on sdio bus: %s\n", sdio_path[i]);
         }
     }
+    if (f == NULL)
+        return -1;
     if (fgets(linebuf, sizeof(linebuf) -1, f))
     {
         ALOGD("read vendor info: %s", linebuf);
-- 
1.8.0

